### ISSUE TITLE
Expose a connection_closed Future / wait_closed Coroutine

### ISSUE THREAD

**Issue by lgrahl** [aaugustin/websockets#469]
I spawn several tasks in the **client** where each of them stops once the connection has been closed (e.g. `while not connection.connection_closed_future.done()`. Currently, I'm accessing the internal (?) future `WebSocketCommonProtocol.connection_lost_waiter` and resolve my future with `WebSocketCommonProtocol.close code`.

It would be great if there were a built-in mechanism to do this. Or perhaps this (or a more fitting mechanism) is already implemented and I just haven't found it?

**Comment by lgrahl** [comment]
I'm not sure which category my code (see worker tasks [1](https://github.com/saltyrtc/saltyrtc-server-python/blob/f4dd6c279b2fdbdca6eff7e9b564cb27bafcebe3/saltyrtc/server/server.py#L543), [2](https://github.com/saltyrtc/saltyrtc-server-python/blob/f4dd6c279b2fdbdca6eff7e9b564cb27bafcebe3/saltyrtc/server/server.py#L567), [3](https://github.com/saltyrtc/saltyrtc-server-python/blob/f4dd6c279b2fdbdca6eff7e9b564cb27bafcebe3/saltyrtc/server/server.py#L600) and [4](https://github.com/saltyrtc/saltyrtc-server-python/blob/f4dd6c279b2fdbdca6eff7e9b564cb27bafcebe3/saltyrtc/server/server.py#L667)) fits into but your suggested solution would definitely fix it. :+1:

**Comment by aaugustin** [comment]
Just to be sure there isn't a misunderstanding...

If you want to do things while the connection is open, `while ws.open` is the way to go (documented [here](https://websockets.readthedocs.io/en/stable/api.html#websockets.protocol.WebSocketCommonProtocol.open)).

If you want to await the moment the connection is closed, then that boolean doesn't do the job. You need a Future. That's what I was discussing above — and it could be a sensible improvement regardless.

**Comment by lgrahl** [comment]
Good point. Seems the code that required awaiting the *closed* future has been factored away at some point. But like you said, even if I don't need it, having such a future is a nice improvement.
